src/main/java/io/stargate/sgv2/jsonapi/exception/mappers/ThrowableToErrorMapper.java

```diff
@@ -30,71 +32,90 @@ public final class ThrowableToErrorMapper {
-              Response.Status.UNAUTHORIZED);
+          return ErrorCode.UNAUTHORIZED_REQUEST
+              .toApiException()
+              .getCommandResultError(message, Response.Status.UNAUTHORIZED);
+          // Driver QueryValidationException -> ErrorCode.INVALID_QUERY
         } else if (throwable instanceof QueryValidationException) {
-          if (message.contains("vector<float,")) { // TODO is there a better way?
+          if (message.contains("vector<float,")) {
             message = "Mismatched vector dimension";
           }
-          fields = Map.of("errorCode", ErrorCode.INVALID_REQUEST.name());
-          return new CommandResult.Error(message, fieldsForMetricsTag, fields, Response.Status.OK);
+          return ErrorCode.INVALID_QUERY
+              .toApiException()
+              .getCommandResultError(message, Response.Status.OK);
+          // Driver Timeout Exception -> ErrorCode.OPERATION_TIMEOUT
         } else if (throwable instanceof DriverTimeoutException
             || throwable instanceof WriteTimeoutException
             || throwable instanceof ReadTimeoutException) {
-          return new CommandResult.Error(message, fieldsForMetricsTag, fields, Response.Status.OK);
-        } else if (throwable instanceof DriverException) {
-          if (throwable instanceof AllNodesFailedException) {
-            Map<Node, List<Throwable>> nodewiseErrors =
-                ((AllNodesFailedException) throwable).getAllErrors();
-            if (!nodewiseErrors.isEmpty()) {
-              List<Throwable> errors = nodewiseErrors.values().iterator().next();
-              if (errors != null && !errors.isEmpty()) {
-                Throwable error =
-                    errors.stream()
-                        .findAny()
-                        .filter(
-                            t ->
-                                t instanceof AuthenticationException
-                                    || t instanceof IllegalArgumentException)
-                        .orElse(null);
-                // connecting to oss cassandra throws AuthenticationException for invalid
-                // credentials connecting to AstraDB throws IllegalArgumentException for invalid
-                // token/credentials
-                if (error instanceof AuthenticationException
-                    || (error instanceof IllegalArgumentException
-                        && (error.getMessage().contains("AUTHENTICATION ERROR")
-                            || error
-                                .getMessage()
-                                .contains(
-                                    "Provided username token and/or password are incorrect")))) {
-                  return new CommandResult.Error(
-                      "UNAUTHENTICATED: Invalid token",
-                      fieldsForMetricsTag,
-                      fields,
-                      Response.Status.UNAUTHORIZED);
-                }
+          return ErrorCode.DRIVER_TIMEOUT
```

Comment:
We really should get more information from exception (e.getMessage()?) to pass along. Otherwise it's 

    Driver timeout on

with nothing describing what actually happens?

Reply:
Ok. As long as that is retained in log output that's fine. You could also pass it with

   .toApiException(message)

but since we do not actually use exception for anything I guess it does not matter.